backend/benefit/common/tests/test_utils.py

```diff
@@ -51,3 +59,72 @@ def test_days360_errors(date1, date2):
 )
 def test_date_range_overlap(start_1, end_1, start_2, end_2, expected_result):
     assert date_range_overlap(start_1, end_1, start_2, end_2) == expected_result
+
+
+def test_get_date_range_end_with_days360():
+
+    assert get_date_range_end_with_days360(date(2022, 1, 1), 0) is None
+    assert (
+        get_date_range_end_with_days360(date(2022, 1, 1), decimal.Decimal("0.01"))
+        is None
+    )
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 1), decimal.Decimal("0.03")
+    ) == date(2022, 1, 1)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 1), decimal.Decimal("0.92")
+    ) == date(2022, 1, 28)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 1), decimal.Decimal("1")
+    ) == date(2022, 1, 31)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 1), decimal.Decimal("1.04")
+    ) == date(2022, 2, 1)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 20), decimal.Decimal("1.38")
+    ) == date(2022, 2, 28)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 30), decimal.Decimal("6.99")
+    ) == date(2022, 8, 30)
+    # special case
+    assert (
+        get_date_range_end_with_days360(date(2022, 2, 28), decimal.Decimal("0.04"))
+        is None
+    )
+
+    if os.getenv("DAYS_360_FULL_TEST") == "1":
+        # the nested loops test a huge number of combinations. Takes a long time to run,
+        # only run this while debugging.
+        for n_days in range(0, 400):
+            print(f"start {n_days}")
+            for initial_day_offset in range(0, 100):
+                for months in [decimal.Decimal(m) / 100 for m in range(0, 1200)]:
```

Comment:
400 * 100 * 1200 = 48 million combinations, ok, so that's infeasible to run on all test runs.

But maybe a randomized but deterministic—i.e. always using the same seed—subset of these could be run normally, e.g. 5 * 5 * 5 = 125 tests or even 10 * 10 * 10 = 1k tests? Or just some of the more important cases and corner cases?

But I'm ok with leaving this as it is now.